File: lab_03_Hibernate/code/src/main/java/org/example/Product.java
package org.example;

import javax.persistence.*;
import java.util.HashSet;
import java.util.Set;

@Entity
public class Product {
    public Product() {}

    public Product(String productName, int unitsOnStock) {
        ProductName = productName;
        UnitsOnStock = unitsOnStock;
    }

    @Id
    @GeneratedValue( strategy = GenerationType.AUTO)
    private int ProductID;

    private String ProductName;
    private int UnitsOnStock;

    public void setSupplier(Supplier supplier) {
        this.supplier = supplier;
        this.supplier.addToProductsGroup(this);
    }

    public String getProductName() {
        return this.ProductName;
    }

    public Supplier getSupplier() {
        return supplier;
    }
//
    @ManyToOne
    private Supplier supplier ;

    public Set<Invoice> getInvoices() {
        return invoices;
    }

    public void addToInvoice(Invoice invoice){
        this.invoices.add(invoice);
        invoice.addProduct(this);
    }

    @ManyToMany(cascade = CascadeType.PERSIST)
    private Set<Invoice> invoices= new HashSet<>();

}

File: lab_02_Entity_Framework/Product.cs
﻿using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace JakubBarberEFProducts
{
    public class Product
    {
        public Product() { 
            this.Invoices = new HashSet<Invoice>();
        }    
        public int ProductID { get; set; }
        public string? ProductName { get; set; }
        public int UnitsOnStock { get; set; }

        //public Supplier? Supplier { get; set; }
        public Supplier? Supplier { get; set; }
        public ICollection<Invoice> Invoices { get; set; }
    }
}
